lib/train/platforms/detect/os_common.rb

```diff
@@ -20,36 +20,37 @@ def winrm?
 
     def unix_file_contents(path)
       # keep a log of files incase multiple checks call the same one
-      return @files[path] unless @files[path].nil?
+      return @files[path] if @files.key?(path)
 
       res = @backend.run_command("test -f #{path} && cat #{path}")
-      @files[path] = res.stdout
       # ignore files that can't be read
-      return nil if res.exit_status != 0
-      res.stdout
+      @files[path] = res.exit_status.zero? ? res.stdout : nil
+      @files[path]
     end
 
     def unix_file_exist?(path)
       @backend.run_command("test -f #{path}").exit_status.zero?
     end
 
+    def uname_call(cmd)
```

Comment:
This is really just `run_command` now, rather than uname-specific... maybe we should change the method name and that way it can be used for other things too?